pydantic/_internal/_model_construction.py

```diff
@@ -72,7 +80,7 @@ def inspect_namespace(  # noqa C901
     for var_name, value in list(namespace.items()):
         if var_name == 'model_config':
             continue
-        elif isinstance(value, all_ignored_types):
+        elif isinstance(value, all_ignored_types) or value.__class__.__module__ == 'functools':
```

Comment:
There is another reference to `all_ignored_types` below, should we make the change there as well? It's only for annotations, which obviously seem less likely to be in `functools`.

I guess you probably already considered this, but anyway, just pointing it out just in case.

Reply:
(Also, to be clear, I think this change is an improvement.)

Reply:
it didn't break anything, so I was lazy, should probably fix.